File: 2021/day01/day01.py
#!/usr/bin/env python3

sample = [ 199, 200, 208, 210, 200, 207, 240, 269, 260,263 ]

def compareElements(input):
    last = None
    for current in sample:
        if last is not None:
            if current > last:
                yield True
            else:
                yield False
        last = current

def part1(list):
    return len([ x for x in  compareElements(list) if x is True])

print ("Sample:")
print (part1(sample))

f = open("input.txt","r")
sample = [ int(x) for x in f.readlines()]

print("Part 1")
print (part1(sample))

File: 2022/day01/day01.rb
#!/usr/bin/env ruby


def day1(f)
  elves = []
  current_elf = 0
  elf_num = 1
  File.foreach(f) { |each_line|
    if each_line == "\n"
      elves.append([elf_num, current_elf])
      elf_num = elf_num + 1
      current_elf = 0
    else
      current_elf += each_line.to_i
    end
  }
  elves.append( [elf_num, current_elf])
  
  return elves.sort_by { |x| x.last }
end

def part1(f)
  elves = day1(f)
  puts elves.last.last
end

def part2(f)
  elves = day1(f).reverse.take(3).map { |x| x.last }.sum
  puts elves
  
end

puts "Part 1"
part1("sample.txt")
part1("input.txt")

puts "Part 2"
part2("sample.txt")
part2("input.txt")